test/libsolidity/SMTCheckerJSONTest.cpp

```diff
@@ -50,6 +54,9 @@ SMTCheckerJSONTest::SMTCheckerJSONTest(string const& _filename, langutil::EVMVer
 		!m_smtResponses.isObject()
 	)
 		BOOST_THROW_EXCEPTION(runtime_error("Invalid JSON file."));
+
+
+	m_shouldRun = ModelChecker::availableSolvers().some();
```

Comment:
Should this be ``if(!ModelChecker::availableSolvers().some()) m_shouldRun = false;``? So that you can also specify evm version restrictions, etc.? I'm not sure, I'd need to look through the new "shouldRun" mechanism.

Also in the future we could still check the query hashes, but just ignore the warning expectations, if there are no solvers, right? But if we want that, I'd put that in some separate issue and not do it right away.

Reply:
The number of hashes changes if there are solvers available or not, which is what happens in the current error.

True, will update re `m_shouldRun`.

Reply:
Ah, ok.